### PR TITLE
PCSX2-WX: Minor changes to MainFrame 

### PR DESCRIPTION
**Pull request by ssakash**
**Summary of Changes**:

* Remove Full/Fast Boot menu items from ``System`` Menu.
* Properly display the boot medium on menu item ``Example: ISO , CDVD , BIOS``
* ~~**Future Change**: #1368 needs to be merged before including this one.~~

**Master Branch**:
Before

**Pull Request Branch**:
MainFrame

### CODE REVIEW

**Comment by refractionpcsx2** [comment]
While I like the change saying Boot ISO/CDVD, I'm not sure making a skip bios option is making it more user friendly, I prefer having the 2 options.  Maybe make fast boot the top option as that will be mostly used.

**Comment by turtleli** [comment]
I prefer 2 separate options as well for the same reason. Though I don't think fast boot and full boot should be swapped.

**Comment by gregory38** [comment]
IMHO It is very confusing.

**Comment by ssakash** [comment]
> I'm not sure making a skip bios option is making it more user friendly, I prefer having the 2 options you can run without having to configure anything (this adds an extra click for little gain). Maybe make fast boot the top option as that will be mostly used.

Is convenience the only reason you don't like the new change ? ``Fast Boot`` and ``Full Boot`` might look confusing at start for new users. also I haven't seen any other emulators having two types of options to boot games. It looks much cleaner to have a separate option to **skip** the logo/splash screen and it's more explicit to the users. so I vote for user friendliness over convenience ;)

> EDIT: Oh - the Linux testing label - the secret is to use a Linux VM/boot :P If you're using a Linux VM you'll need to use the legacy GSdx plugin (if mesa is new enough it'll work).

Well, I do have a slackware distro on dual boot (along with Windows 10) but just too lazy to learn how to use it.  (Yes, I'm a complete noob on linux stuffs)

It took lots of dedication to even install it lol, [Thank you spatry !](https://www.youtube.com/watch?v=DdwZylEQcQ0)

> IMHO It is very confusing.

Please be more verbose :P

**Comment by Blackbird88** [comment]
Honestly I like the Skip BIOS. Dolphin does it too with GameCube and I fail to see how it's confusing.

**Comment by refractionpcsx2** [comment]
> Dolphin does it too with GameCube and I fail to see how it's confusing.

As much respect as I have for the Dolphin team and how impressive their emulator is, I'm kinda getting sick of the "Dolphin does it" argument as a reason to why we should do it. That isn't a valid reason.

**Comment by Blackbird88** [comment]
It was just example and regardless of it I still find it better solution. Full/Fast boot are confusing.

**Comment by FlatOutPS2** [comment]
Changing it something akin to this should make it easier for users while keeping both options:
Boot ... - Skip BIOS (Preferred)
Boot ... - Full sequence

**Comment by refractionpcsx2** [comment]
What about just changing Full Boot to BIOS Boot? then Fast Boot can stay the same.

**Comment by FlatOutPS2** [comment]
> What about just changing Full Boot to BIOS Boot? then Fast Boot can stay the same.

That might trick some people into thinking it will boot into the bios, not the game.

**Comment by refractionpcsx2** [comment]
It's very difficult to come up with a name which describes what it does without sounding confusing or overly complicated

**Comment by gregory38** [comment]
```
Normal boot
No region check boot
```
Is there any valid reason to use fast boot ?

**Comment by mirh** [comment]
Indeed, we shouldn't trick people into believing "it just skips bios part". 
You miss game localization then. 

Full sounds better than normal in this regard.

**Comment by FlatOutPS2** [comment]
> Is there any valid reason to use fast boot ?

For new users it's better to at least indicate it's faster than full boot.
`No region check boot (faster)`

**Comment by gregory38** [comment]
My issue with fast is that people could think it would make the emulation faster whereas it saves a couple of seconds at startup.

**Comment by FlatOutPS2** [comment]
Then it's better to go back to skip instead of no, so that there's a better indication of it saving some time.
` Boot ... - Skip region check`

**Comment by gregory38** [comment]
Well it was just a random idea. I like your proposal. This way people can understand the intent.

**Comment by ramapcsx2** [comment]
I'm also for keeping 2 entries but I like the suggestion
Boot ... - Skip BIOS (Preferred)
Boot ... - Full sequence

Maybe I'd leave out the "preferred". The naming already suggests that "Skip Bios" might be quick and "Full sequence" might be complete. It's good :)

**Comment by ssakash** [comment]
FWIW, I opened a [poll](https://www.facebook.com/groups/98483509559/permalink/10154387645704560/) on the FB group and most of the people seemed to like the ``SKIP Bios`` option. 

**Poll Results** :
``Add "Skip BIOS" option and remove Full/Fast Boot`` - **14 Votes**
``Keep the Full/Fast Boot options`` - **6 Votes**

> Indeed, we shouldn't trick people into believing "it just skips bios". 

It's not about tricking people, but rather using a layman/short term to make them understand about it better. **Example**: ``Internal Resolution``

and it also partially skips the content of the BIOS (Kinda, I guess) as the initial video mode when you get the splash screen is something different from ``NTSC/PAL/VESA/HDTV`` video modes. There's a special video mode reserved for the BIOS stuffs (``DVD NTSC/PAL``). 

The video mode gets triggered on the splash screen and also when you enter the BIOS for ``System Configuration / Memory Card browser``,

> Then it's better to go back to skip instead of no, so that there's a better indication of it saving some time.
>Boot ... - Skip region check

maybe we could add a helper text along with the ``Skip BIOS`` option stating that ``The option skips the regional lockout process when enabled `` ?

Also , could someone give any good reasons (excluding convenience) on why prefer two menu items over one check item and a menu item ? I think the latter looks much cleaner. also this way we don't have a long text on the menu item.

**Comment by gregory38** [comment]
```
and it also partially skips the content of the BIOS (Kinda, I guess)
```
The trick is that people use their own definition of the BIOS ;) Technically bios is just the stuff to boot the kernel. You still need it (but it could be reduced to something very small). And you still need the kernel ;)
The part that is skipped is Sony's binary loader that does region checking (and others stuff). Instead the kernel will directly load the game binary.

The question is what is the purpose of this option? Booting faster ? region quirk ? Others.

```
Also , could someone give any good reasons (excluding convenience) on why prefer 
two menu items over one check item and a menu item ? I think the latter looks much 
cleaner. also this way we don't have a long text on the menu item.
```
I don't like it because you have interaction between option and the menu behavior. And a side note, nobody understand what skip bios really means.

Hum on the poll, I'm curious if people are biased toward others emulators ;)

**Comment by FlatOutPS2** [comment]
Polls are known to get different answers based on how you ask the question, formulation of the options and the options included. That poll wasn't very clear and 22 voters isn't a good sample of PCSX2 users.

**Comment by mirh** [comment]
Also, if we really have to make something preferred or advised, it should be normal booting.

**Comment by FlatOutPS2** [comment]
There is another option to consider. Leaving just the single `Boot ...` and choose full / fast based on the game database(possibly including a setting somewhere to switch to the old manual way). Most games don't suffer any disadvantages from using the fast/ skip option.

**Comment by ramapcsx2** [comment]
@FlatOutPS2 that's bad for developers though. For example: Full boot will always disable EE DECI2 messages from the games. Quick booted titles will displays more debug information.

**Comment by ssakash** [comment]
> Polls are known to get different answers based on how you ask the question, formulation of the options and the options included. That poll wasn't very clear and 22 voters isn't a good sample of PCSX2 users.

It has only been 5 hours. Maybe more people would vote after some time,

> Also, if we really have to make something preferred or advised, it should be normal booting. 

Yeah, I agree with that.

> I don't like it because you have interaction between option and the menu behavior. 

Honestly I fail to understand how it's bad (or) not preferable.

> And a side note, nobody understand what skip bios really means.

well, I used something like that since I couldn't describe it any shorter. maybe ``Skip Regional Lockout`` ? (Technically it's the right term but might not easily understood by normal users..)

**Comment by FlatOutPS2** [comment]
> It has only been 5 hours. Maybe more people would vote after some time

That won't make the poll much clearer and it's stil only the two options to choose from. They might just be picking the least disliked instead of something they would actually prefer.

@ramapcsx2 That's what the option to switch to old format is for. For normal use that doesn't really matter, and the old format can be default for dev/debug builds.

**Comment by willkuer** [comment]
> The question is what is the purpose of this option? Booting faster ? region quirk ? Others.

Region lock is really a strong point for fast boot. If you remove this option you will get hundreds of threads of type 'game is marked as playable but doesn't work'.

IMO the sony intro is one of the dumbest idea's ever. As if you would forget that you are using a sony playstation 2 because the intro is not shown...

I never use FullBoot.

**Comment by ramapcsx2** [comment]
Boot ... - Skip BIOS
Boot ... - Full sequence

is really good :)
It preserves all the functionality. It's clearer than now.

**Comment by ssakash** [comment]
>Boot ... - Skip BIOS
Boot ... - Full sequence

>is really good :)
It preserves all the functionality. It's clearer than now.

I disagree, it adds too much text into the menu item making the GUI look a bit awkward. the idea is to have something short, clear and concise (something self-explanatory). also apparenty @gregory38 found ``Skip BIOS`` confusing.

> That won't make the poll much clearer and it's stil only the two options to choose from. They might just be picking the least disliked instead of something they would actually prefer.

They could still add an extra option on the poll if they have a better idea.

**Comment by mirh** [comment]
It's not only about devs! You can't even play in a language different than english in many games with fast boot. 
...
*Besides*, what's wrong with the actual layout in the first place?

**Comment by ramapcsx2** [comment]
As worded now, we have 15 letters. The new suggestion is 21 letters, if we use the full version with all the dots and spaces. Skip Bios explains itself when you see the other option, which is "Full sequence".
Your suggestion with the "Skip BIOS" is acceptable, though I'd hate ticking that box on and off, I can see a benefit for regular users. On the other hand, the whole process is still not a whole lot better.. :p

**Comment by FlatOutPS2** [comment]
boot

Doesn't look awkward to me, even the longest version is only just longer than `Enable Widescreen Patches` in the same menu, and the menu width is already more than enough.

**Comment by gregory38** [comment]
Please don't add mess in the game db. The file already clutter the git history (and it isn't a good solution anyway).

I think, it would be better to define the intent of the option and to emphasize the behavior. So far
* `Fast boot` used for region locking. The fact that it is faster to start the game is a nice side effect IMHO. Honestly, compare the time to start the game and the time the game will actually be played (except tester). People save nothing in their tight schedule.
* `Full` used for the full experience.

So I personally prefer Flatout proposal.

**Comment by ssakash** [comment]
@willkuer 

> Ok. so I agree with him. Dev's can just enable the manual gamefix 'FullBoot' if needed.

I'm not sure whether it should be placed on gamefix tab as most of the gamefixes are just hacks made to solve a certain problem whereas ``FullBoot`` is _not actually a hack_.

@FlatOutPS2 

I still don't like the long text and the double usage of ``Boot ISO``. maybe it's just me who doesn't like your proposal. I'll ping for opinion from @avih , he's the expert in these GUI design stuffs after all.

also you restarted the Travis/Appveyor build process by closing/reopening the Pull request  :P

**Comment by DPersonalized** [comment]
How about:
Boot ISO (With BIOS)
Boot ISO (Skip BIOS)

**Comment by mirh** [comment]
How about

    Boot ISO - Complete
    Boot ISO - Fast hack

**Comment by turtleli** [comment]
If longer text makes it clearer then longer text is preferable. FlatOut's screenshot looks reasonable to me.

> also you restarted the Travis/Appveyor build process by closing/reopening the Pull request

By the way, how do you push your git changes? Is it automatic or manual? Sometimes I notice there are 3-4 builds in the AppVeyor queue for some of your PRs and I wonder whether it could be avoided.

**Comment by ssakash** [comment]
> If longer text makes it clearer then longer text is preferable. FlatOut's screenshot looks reasonable to me.

Hmm, the double ``Boot ISO`` doesn't bug you ? Honestly I prefer short and concise text over longer ones.

I feel it'd be better to separate hacks like ``Fast Boot``  with something like a check item (or) as a checkbox in the speedhacks panel , that's my reasoning for the separation. Generally it would be ideal to have a single menu item to allow booting of the medium instead of multiple ones.


> By the way, how do you push your git changes? Is it automatic or manual? Sometimes I notice there are 3-4 builds in the AppVeyor queue for some of your PRs and I wonder whether it could be avoided.

I just do ``git push -f``

I think the extra builds are probably from my commits for rewording the commit messages / fixing trailing white spaces. (Maybe ? )

**Comment by refractionpcsx2** [comment]
One thought on why the words "skip bios" aren't making me happy, i suspect people will think you don't need a bios if you "skip it" causing annoyance :P

**Comment by ssakash** [comment]
As I said, I don't mind changing it into something like ``Skip Regional Lockout`` , I just chose ``Skip BIOS`` since it was shorter and thought it might be more explicit to the users.

**Comment by turtleli** [comment]
> Hmm, the double Boot ISO doesn't bug you ? Honestly I prefer short and concise text over longer ones.

Not really. I suppose what people might actually want are buttons... then people wouldn't mind so much if the menu was 1 option and 1 checkbox, 2 options, sub menu etc. But yeah GUI.

> I think the extra builds are probably from my commits for rewording the commit messages / fixing trailing white spaces. (Maybe ? )

It looks more like push, rebase, push.

**Comment by ssakash** [comment]
> It looks more like push, rebase, push.

Yeah, I do it sometimes. unless I'm wrong, that's the only way to prevent merge conflicts.

**Comment by ramapcsx2** [comment]
This is good. Isn't it? :)

**Comment by ssakash** [comment]
Isn't that the same thing [FlatOut posted](https://github.com/PCSX2/pcsx2/pull/1381#issuecomment-222963059) ? 

I honestly don't like it for the reasons stated over [here](https://github.com/PCSX2/pcsx2/pull/1381#issuecomment-222965738) but I'm inclining towards the majority. (***waiting for the results of the poll*** ). If anyone thinks the poll at FB doesn't look promising then feel free to open one at the forum.

**Comment by ramapcsx2** [comment]
It's the same, yes. I'm aiming for a nice compromise :)
Doing polls is useless unless you give people all the information they need to know, and in a non-biased way. Doing a correct poll is hard.

**Comment by refractionpcsx2** [comment]
considering the 3rd most popular poll result is "release a new stable version", i wouldn't say it's going very well :P The options are also limited to add skip bios option and remove old ones, or leave it as is. it would have been better to get a range of ideas in, propose them all, then do the poll.

**Comment by tony971** [comment]
Just my two cents as a layman: I understand "skip bios" as skipping the system boot screen directly into the game. Whether or not technically correct, this is what I think of it as. And this is based off of mGBA, mednafen psx, and other emulators, not just Dolphin. It's a commonly used term.

**Comment by FlatOutPS2** [comment]
Instead of skip region check or skip BIOS another option would be `Boot ... Short sequence`, and then explain about skipping the bios region check in the help text.

**Comment by refractionpcsx2** [comment]
I don't like the use of "Sequence" that means even less to the user than skip bios, they know nothing of the "sequence" involved.

**Comment by mirh** [comment]
Full CDVD/ISO boot, short CDVD/ISO boot?

And if not short, we should think which between that, quick and fast is better.

**Comment by i30817** [comment]
Myself i thought the original idea was fine. I don't use the bios at all, especially now that tools like the memcard save manager exists.

Since i have zero influence on what dolphin devs want to do however, this feels fairly useless now. Boot CDVD (full) and (fast) are fine.

**Comment by refractionpcsx2** [comment]
> Since i have zero influence on what pcsx2 devs want to do however, this feels fairly useless now.

everybody has influence, if the masses want something changed, then we will change it, as long as it's feasible.

The BIOS does get used for some things, for example if you want to boot with no DVD or if you need to maintain memorycards, deleting things or formatting etc.

**Comment by ssakash** [comment]
Can I just close this PR ? no one on the team seems to like this change anyway.

Also, The [poll was kinda ruined](http://ultraimg.com/images/poll4aab3.png) after squall added that ``Delete this poll, eat cookies`` option...

**Comment by FlatOutPS2** [comment]
I think everyone does like the ISO/CDVD/Boot indicator being added.

**Comment by ssakash** [comment]
Removed my first commit, only the ISO/CDVD/BIOS indicator change for the menu item is left.

**Comment by refractionpcsx2** [comment]
Looking at these options, It would be good if "no disc" is selected that you remove the fast boot option from the menu as it is invalid in that scenario, so you are only left with "Boot BIOS (full)" which you can of course remove the (full) bit from the end :)

**Comment by ssakash** [comment]
> Looking at these options, It would be good if "no disc" is selected that you remove the fast boot option from the menu as it is invalid in that scenario, so you are only left with "Boot BIOS (full)" which you can of course remove the (full) bit from the end :)

>Or replace the bracketed part with (No Disc), that would make sense.

Made the suggested changes.

File: pcsx2/gui/MainMenuClicks.cpp [hunk 1]
```diff
@@ -368,13 +368,13 @@ void MainEmuFrame::EnableCdvdPluginSubmenu(bool isEnable)
...
 {
-	CDVD_SourceType newsrc = CDVDsrc_NoDisc;
+	CDVD_SourceType newsrc = CDVD_SourceType::CDVDsrc_NoDisc;
 
 	switch( event.GetId() )
 	{
-		case MenuId_Src_Iso:	newsrc = CDVDsrc_Iso;		break;
-		case MenuId_Src_Plugin:	newsrc = CDVDsrc_Plugin;	break;
-		case MenuId_Src_NoDisc: newsrc = CDVDsrc_NoDisc;	break;
+	    case MenuId_Src_Iso:	newsrc = CDVD_SourceType::CDVDsrc_Iso;		break;
```
Comment (turtleli) [inline pcsx2/gui/MainMenuClicks.cpp]: Mixed whitespace.

File: pcsx2/gui/MainFrame.cpp [hunk 2]
```diff
@@ -643,31 +643,65 @@ void MainEmuFrame::ApplyCoreStatus()
...
+			{
+				cdvd->SetItemLabel(_("Reboot BIOS"));
+				cdvd->SetHelp(_("Hard reset of the active VM."));
+				m_menuSys.Remove(cdvd2);
+			}
+			else
+			{
+				cdvd->SetItemLabel(_("Boot BIOS"));
+				cdvd->SetHelp(_("Boot the VM without any source media"));
+				m_menuSys.Remove(cdvd2);
```
Comment (turtleli) [inline pcsx2/gui/MainFrame.cpp]: Memleak - `Remove` doesn't destroy the menu item.

File: pcsx2/gui/MainFrame.cpp [hunk 3]
```diff
@@ -643,31 +643,65 @@ void MainEmuFrame::ApplyCoreStatus()
...
-			cdvd->SetHelp(_("Hard reset of the active VM."));
-		}
-		else
-		{
-			cdvd->SetItemLabel(_("Boot CDVD (full)"));
-			cdvd->SetHelp(_("Boot the VM using the current DVD or Iso source media"));
-		}	
+		m_menuSys.Prepend(cdvd2);
+		m_menuSys.Insert(1, MenuId_Boot_CDVD2);
+		cdvd2 = menubar.FindItem(MenuId_Boot_CDVD2);
```
Comment (turtleli) [inline pcsx2/gui/MainFrame.cpp]: Asserts in debug mode.

`Prepend` doesn't do anything useful, and `Insert` returns the inserted wxMenuItem, so the statement afterwards is somewhat useless.

File: pcsx2/gui/MainFrame.cpp [hunk 4]
```diff
@@ -57,14 +57,14 @@ void MainEmuFrame::UpdateIsoSrcSelection()
 
 	switch( g_Conf->CdvdSource )
 	{
-		case CDVDsrc_Iso:		cdsrc = MenuId_Src_Iso;		break;
-		case CDVDsrc_Plugin:	cdsrc = MenuId_Src_Plugin;	break;
-		case CDVDsrc_NoDisc:	cdsrc = MenuId_Src_NoDisc;	break;
+		case  CDVD_SourceType::CDVDsrc_Iso:		cdsrc = MenuId_Src_Iso;		break;
+		case  CDVD_SourceType::CDVDsrc_Plugin:	cdsrc = MenuId_Src_Plugin;	break;
+		case  CDVD_SourceType::CDVDsrc_NoDisc:	cdsrc = MenuId_Src_NoDisc;	break;
```
Comment (turtleli) [inline pcsx2/gui/MainFrame.cpp]: Double spaces after case.

File: pcsx2/gui/MainFrame.cpp [hunk 5]
```diff
@@ -594,16 +594,16 @@ void MainEmuFrame::OnActivate( wxActivateEvent& evt )
...
-	wxMenuItem* cdvd2	= menubar.FindItem( MenuId_Boot_CDVD2 );
+	wxMenuItem* cdvd2 = menubar.FindItem( MenuId_Boot_CDVD2 );
 	wxMenuItem* restart	= menubar.FindItem( MenuId_Sys_Restart );
+	CDVD_SourceType source = g_Conf->CdvdSource;
 
 	// [TODO] : Ideally each of these items would bind a listener instance to the AppCoreThread
 	// dispatcher, and modify their states accordingly.  This is just a hack (for now) -- air
 
-	bool vm = SysHasValidState();
+	bool Active = SysHasValidState();
```
Comment (turtleli) [inline pcsx2/gui/MainFrame.cpp]: Maybe `vmIsActive` would be clearer?

File: pcsx2/CDVD/CDVDaccess.cpp [hunk 6]
```diff
@@ -354,8 +354,9 @@ bool DoCDVDopen()
 	// question marks if the filename is another language.
 	// Likely Fix: Force new versions of CDVD plugins to expect UTF8 instead.
 
-	int ret = CDVD->open( !m_SourceFilename[m_CurrentSourceType].IsEmpty() ?
-		static_cast<const char*>(m_SourceFilename[m_CurrentSourceType].ToUTF8()) : (char*)NULL
+	int CurrentSourceType = static_cast<int>(m_CurrentSourceType);
+	int ret = CDVD->open( !m_SourceFilename[CurrentSourceType].IsEmpty() ?
+		static_cast<const char*>(m_SourceFilename[CurrentSourceType].ToUTF8()) : (char*)NULL
```
Comment (turtleli) [inline pcsx2/CDVD/CDVDaccess.cpp]: Not your code, but is casting null necessary?

File: pcsx2/gui/MainMenuClicks.cpp [hunk 7]
```diff
@@ -133,7 +133,7 @@ wxWindowID SwapOrReset_Iso( wxWindow* owner, IScopedCoreThread& core_control, co
 {
 	wxWindowID result = wxID_CANCEL;
 
-	if( (g_Conf->CdvdSource == CDVDsrc_Iso) && (isoFilename == g_Conf->CurrentIso) )
+	if( (g_Conf->CdvdSource == CDVD_SourceType::CDVDsrc_Iso) && (isoFilename == g_Conf->CurrentIso) )
```
Comment (turtleli) [inline pcsx2/gui/MainMenuClicks.cpp]: Again, not your code, but are the additional `()` necessary?

File: pcsx2/gui/MainMenuClicks.cpp [hunk 8]
```diff
@@ -158,12 +158,12 @@ wxWindowID SwapOrReset_Iso( wxWindow* owner, IScopedCoreThread& core_control, co
...
 	}
 
-	g_Conf->CdvdSource = CDVDsrc_Iso;
+	g_Conf->CdvdSource = CDVD_SourceType::CDVDsrc_Iso;
 	SysUpdateIsoSrcFile( isoFilename );
 	if( result == wxID_RESET )
 	{
 		core_control.DisallowResume();
-		sApp.SysExecute( CDVDsrc_Iso );
+		sApp.SysExecute(CDVD_SourceType::CDVDsrc_Iso );
```
Comment (turtleli) [inline pcsx2/gui/MainMenuClicks.cpp]: Spacing around parenthesis is inconsistent - either there are spaces on each side or there are none. Same applies to some of the other changes.

**Comment by ramapcsx2** [comment]
turtleli you going to be chief code style police? :D

**Comment by refractionpcsx2** [comment]
I don't know if that's better or worse than Grammar Nazi :P

**Comment by FlatOutPS2** [comment]
> I don't know if that's better or worse than Grammar Nazi :P

You don't know what's worse, police or nazi? :S

**Comment by refractionpcsx2** [comment]
> You don't know what's worse, police or nazi? :S

different kind of nazi's ;)

File: pcsx2/gui/MainFrame.cpp [hunk 9]
```diff
@@ -643,31 +643,63 @@ void MainEmuFrame::ApplyCoreStatus()
...
-		{
-			cdvd->SetItemLabel(_("Reboot CDVD (full)"));
-			cdvd->SetHelp(_("Hard reset of the active VM."));
-		}
-		else
-		{
-			cdvd->SetItemLabel(_("Boot CDVD (full)"));
-			cdvd->SetHelp(_("Boot the VM using the current DVD or Iso source media"));
-		}	
+		cdvd2 = m_menuSys.Insert(1, MenuId_Boot_CDVD2);
```
Comment (turtleli) [inline pcsx2/gui/MainFrame.cpp]: This still trips an assertion in a debug build and can crash PCSX2 as well.

**Comment by ssakash** [comment]
> This still trips an assertion in a debug build and can crash PCSX2 as well.

Well, I just changed 

``cdvd2 = m_menuSys.Insert(1, MenuId_Boot_CDVD2);``

**into**

``cdvd2 = m_menuSys.Insert(1, MenuId_Boot_CDVD2, "Turtleli");``

and **Whoosh** , Assert is no longer triggered ! All hail **lord turtleli** :smile:

File: pcsx2/gui/MainFrame.cpp [hunk 10]
```diff
@@ -643,31 +643,65 @@ void MainEmuFrame::ApplyCoreStatus()
...
-			cdvd->SetHelp(_("Hard reset of the active VM."));
-		}
-		else
-		{
-			cdvd->SetItemLabel(_("Boot CDVD (full)"));
-			cdvd->SetHelp(_("Boot the VM using the current DVD or Iso source media"));
-		}	
+		// WX throws an assert when a non stock menu item is inserted without a label text as a parameter on Insert()
+		// Let's add a placeholder text just to avoid an assert.
+		cdvd2 = m_menuSys.Insert(1, MenuId_Boot_CDVD2, "Turtleli");
```
Comment (turtleli) [inline pcsx2/gui/MainFrame.cpp]: Missing silent k and x - `tkurtxleli`.

Would it not be better to rearrange things so you don't need to use some random text?

Comment (ssakash) [inline pcsx2/gui/MainFrame.cpp]: > Would it not be better to rearrange things so you don't need to use some random text?

Did I catch your drift ?

```
 case CDVD_SourceType::CDVDsrc_Iso:
			if (vmIsActive)
			{
				cdvd->SetItemLabel(_("Reboot ISO (full)"));
				cdvd->SetHelp(_("Hard reset of the active VM."));
               if(cdvd2)
				    cdvd2->SetItemLabel(_("Reboot ISO (fast)"));
               else
                    cdvd2 = m_menuSys.Insert(1, MenuId_Boot_CDVD2, "Reboot ISO (fast)");
				cdvd2->SetHelp(_("Reboot using fast BOOT (skips splash screens)"));
			}
```

I prefer the current code over something like the code snippet present above.

Comment (turtleli) [inline pcsx2/gui/MainFrame.cpp]: Personally I was thinking of something like:
```
wxString full_label;
wxString full_help;
wxString fast_label;
wxString fast_help;

//
// Set label and help strings
//

cdvd->SetItemLabel(full_label);
cdvd->SetHelp(full_help);

if (source == CDVD_SourceType::CDVDsrc_NoDisc) {
	if (cdvd2)
		m_menuSys.Destroy(cdvd2);
} else {
	if (cdvd2) {
		cdvd2->SetItemLabel(fast_label);
		cdvd2->SetHelp(fast_help);
	} else {
		m_menuSys.Insert(1, MenuId_Boot_CDVD2, fast_label, fast_help);
	}
}
```
It'd save copy paste replication.

Comment (ssakash) [inline pcsx2/gui/MainFrame.cpp]: Update the code to something like that, I liked ``cdvd2 = m_menuSys.Insert(1, MenuId_Boot_CDVD2, "Turtleli");`` better :cry:

File: pcsx2/gui/MainFrame.cpp [hunk 11]
```diff
@@ -643,32 +644,55 @@ void MainEmuFrame::ApplyCoreStatus()
...
 	}
 
-	if( cdvd )
+	wxString Source_Label[static_cast<int>(CDVD_SourceType::NumberofSources)] = {"ISO" , "CDVD" , "BIOS"};
+	wxString Reboot_Boot_Indicator = (vmIsActive) ? "Reboot" : "Boot";
+	wxString ItemLabel = Reboot_Boot_Indicator + " " + Source_Label[static_cast<int>(source)];
+	wxString Fast_Reboot_HelpText = "Reboot using fast BOOT (skips splash screens)";
+	wxString Fast_Boot_HelpText = "Use fast boot to skip PS2 startup and splash screens";
+	wxString Full_Boot_HelpText[static_cast<int>(CDVD_SourceType::NumberofSources)];
+	wxString Full_Reboot_HelpText = "Hard reset of the active VM";
```
Comment (turtleli) [inline pcsx2/gui/MainFrame.cpp]: You removed string translations. 

I'm going to suggest you either use `thisNamingConvention` or `this_naming_convention` for variables. See  https://github.com/PCSX2/pcsx2/pull/953#issuecomment-154707005

File: pcsx2/gui/MainFrame.cpp [hunk 12]
```diff
@@ -643,32 +644,55 @@ void MainEmuFrame::ApplyCoreStatus()
...
+			cdvd->SetItemLabel(ItemLabel + " (f&ull)");
+
+		if (vmIsActive)
+			cdvd->SetHelp(Full_Reboot_HelpText);
+		else
+			cdvd->SetHelp(Full_Boot_HelpText[static_cast<int>(source)]);
 	}
 
-	if( cdvd2 )
+	if (cdvd2 || Destroyed_CDVD2)
```
Comment (turtleli) [inline pcsx2/gui/MainFrame.cpp]: WHAT.

Comment (ssakash) [inline pcsx2/gui/MainFrame.cpp]: Care to explain what's wrong here ? ``WHAT`` isn't helping much.

Comment (refractionpcsx2) [inline pcsx2/gui/MainFrame.cpp]: Destroyed_CDVD2 will never ever be true in that instance. It is reset at the start at the function to false and only set to true after that providing cdvd2 is true, but then it's never used.

That's one thing anyway :P

Comment (ssakash) [inline pcsx2/gui/MainFrame.cpp]: > Destroyed_CDVD2 will never ever be true in that instance. It is reset at the start at the function to false and only set to true after that providing cdvd2 is true, but then it's never used.

No, It'll only get initialized once. I'm not foolish enough to make a mistake like that. Knowing turtleli he probably didn't like the variable name (or) something :P

Comment (turtleli) [inline pcsx2/gui/MainFrame.cpp]: It's equivalent to `if (cdvd2 || !cdvd2)`. \*sigh\*

Comment (danilaml) [inline pcsx2/gui/MainFrame.cpp]: @refractionpcsx2 http://en.cppreference.com/w/cpp/language/storage_duration#Static_local_variables

Comment (ssakash) [inline pcsx2/gui/MainFrame.cpp]: > It's equivalent to if (cdvd2 || !cdvd2). *sigh*

No, you're wrong. ``Destroyed_CDVD2`` tracks only the cases where we explicitly destroyed the menu item. What if something else happens and cdvd2 dereferences null ? in that case the code won't enter the true branch.

Comment (turtleli) [inline pcsx2/gui/MainFrame.cpp]: So what else can happen to cdvd2? And why is it necessary to track whether you destroyed it explicitly or not?

Comment (ssakash) [inline pcsx2/gui/MainFrame.cpp]: > So what else can happen to cdvd2? And why is it necessary to track whether you destroyed it explicitly or not?

Hmm, Nice point :P 

Maybe I got a bit overprotective on that one and the code went into a superfluous state. I'll fix the translation and remove the unnecessary tracking of explicit destruction tomorrow. way past my bed time right now.

**Pull request closed without merging**